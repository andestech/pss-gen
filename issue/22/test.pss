/*!
@weakgroup      issue
@{
@file
@author         Luther Lee
@data           2024/07/03
@brief          Test issue 22.


---------------------------------------------------------------------------------------------------

### Test Command
@code{.unparsed}
make
@endcode

@cond
*/
component pss_top {
    action root_a {
        activity {
            do invertSingleBit with { val in [0]; };
            do invertSingleBit with { val in [1]; };
        }
    }

    action invertSingleBit {
        rand bit val;
        bit val_n;

        exec post_solve {
            val_n = ~val;
        }

        exec body ASM = """invertSingleBit: {{val}} -> {{val_n}}""";
    }
}

/**@}*/
