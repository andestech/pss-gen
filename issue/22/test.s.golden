invertSingleBit: 0 -> 1
invertSingleBit: 1 -> 0
